### DISCUSSION TITLE
FFT spectrogram visulization

### DISCUSSION THREAD

**Discussion by dizcza** [sukesh-ak/ESP32-TUX#23] [General]
Hi there,
has anyone tried running FFT spectrograms on WT32-SC01 devices? Such large screens would make spectrum visualization more informative and pleasant to look at. I'm sure in the future many others will be looking for it too.
I did a search on the web and found something similar here https://github.com/atomic14/esp32-tft-audio-monitor. This port uses a well-known arduino TFT_eSPI lib. I don't know if it's available for ESP-IDF.
I may try porting it here or give up completely and switch to devices with smaller screens that already have this feature implemented (like M5Stack). But if someone is paying attention to this topic too, I'll be more tempted to have a try with WT32.

**Comment by sukesh-ak** [comment]
> Hi there, has anyone tried running FFT spectrograms on WT32-SC01 devices? Such large screens would make spectrum visualization more informative and pleasant to look at. I'm sure in the future many others will be looking for it too. I did a search on the web and found something similar here https://github.com/atomic14/esp32-tft-audio-monitor. This port uses a well-known arduino TFT_eSPI lib. I don't know if it's available for ESP-IDF. I may try porting it here or give up completely and switch to devices with smaller screens that already have this feature implemented (like M5Stack). But if someone is paying attention to this topic too, I'll be more tempted to have a try with WT32.

Should be pretty easy to have it.
Audio portions are available as part of ESP-ADF and graph can be done already using just LovyanGFX or with LVGL.
